presto-main/src/main/java/com/facebook/presto/memory/ClusterMemoryManager.java

```diff
@@ -260,16 +260,23 @@ public synchronized void process(Iterable<QueryExecution> runningQueries, Suppli
         clusterUserMemoryReservation.set(totalUserMemoryBytes);
         clusterTotalMemoryReservation.set(totalMemoryBytes);
 
-        if (!(lowMemoryKiller instanceof NoneLowMemoryKiller) &&
+        boolean killOnOOMDelayPassed = nanosSince(lastTimeNotOutOfMemory).compareTo(killOnOutOfMemoryDelay) > 0;
+        boolean lastKilledQueryGone = isLastKilledQueryGone();
+        boolean shouldCallOomKiller = !(lowMemoryKiller instanceof NoneLowMemoryKiller) &&
                 outOfMemory &&
                 !queryKilled &&
-                nanosSince(lastTimeNotOutOfMemory).compareTo(killOnOutOfMemoryDelay) > 0) {
-            if (isLastKilledQueryGone()) {
-                callOomKiller(runningQueries);
-            }
-            else {
-                log.debug("Last killed query is still not gone: %s", lastKilledQuery);
-            }
+                killOnOOMDelayPassed &&
+                lastKilledQueryGone;
+
+        if (shouldCallOomKiller) {
+            callOomKiller(runningQueries);
+        }
+        else {
+            log.debug("OOM killer is not called. Cluster is out of memory: %s, query killed: %s, kill on OOM delay passed: %s, last killed query gone: %s",
```

Comment:
Are we logging this even when cluster is not out of memory? Is this necessary?

Reply:
Good point! I don't think we need to log it whenever the cluster is not OOM. We can log it when the cluster is OOM but the oom killer is not called.